File: 20221013/모범답안/20221013_06/MaxBetween.js
/**
 * @param arr {number[]}
 * @param a {number}
 * @param b {number}
 * @returns {number}
 */
function solution(arr, a, b) {
  let curMax = null
  arr.forEach(num => {
    const isIn = a <= num && num <= b
    const isMax = curMax === null || curMax < num
    curMax = isIn && isMax ? num : curMax
  })
  return curMax === null ? -1 : curMax
}

export default solution

File: 20221012/모범답안/20221012_06/MaxBetween.py
def solution(arr, a, b):
    """
    :param arr: int[]
    :param a: int
    :param b: int
    :return: int
    """

    cur_max = None
    for num in arr:
        is_in = a <= num <= b
        is_max = cur_max is None or cur_max < num
        cur_max = num if is_in and is_max else cur_max

    return -1 if cur_max is None else cur_max
